### ISSUE TITLE
Example commands don't work with default config (possible readme improvement)

### ISSUE THREAD

**Issue by sandersantema** [wbthomason/packer.nvim#217]
This might be a quirk of my setup but to get the `use` and `use_rocks` methods working I need:
```lua
return require('packer').startup(function(use,use_rocks)
```
Instead of:
```lua
return require('packer').startup(function()
```

I haven't directly opened a pr given that I'm not sure if this is a quirk of my setup, but if it's not a quirk I suggest changing the example config in `README.MD`.

**Comment by wbthomason** [comment]
That example should work/has worked for me when I've tested it in the past. What errors do you see?

For reference, this is where we ensure that `use` and `use_rocks` are in the `fenv` of the function passed to startup, whether or not they're taken as arguments:
https://github.com/wbthomason/packer.nvim/blob/993eaad39b3ead6dbe1cbe6428e8605c5a63dd2f/lua/packer.lua#L465-L468

**Comment by sandersantema** [comment]
Ah indeed, apparently it's not the best idea to blindly trust my language server :) I do however wonder how you get around the undefined global problem. Looking trough your personal dotfiles I don't see any special globals configured for `sumneko_lua`.

**Comment by wbthomason** [comment]
I don't currently use the `startup` style, so I haven't run into that with `sumneko` myself. That said, I do customize `sumneko_lua` with some globals (though it's really just the `vim.*` stuff):
https://github.com/wbthomason/dotfiles/blob/6a4eb37df63953e03891ba04cf6ba2b4ec4270e8/neovim/.config/nvim/lua/config/lsp.lua#L122-L127

In actual use of `packer`, we get around the undefined global issue by using `setfenv` for the user function. That allows us to modify the binding environment of the function to define `use` and `use_rocks`.

**Comment by akinsho** [comment]
Another way to avoid the `use` global lsp error is to use the function passed in explicitly to the startup function, that way you don't have to put a special override in the sumneko settings. Although I really wish that `sumneko` could somehow figure out that it is available within that scope.

**Comment by sandersantema** [comment]
> Another way to avoid the `use` global lsp error is to use the function passed in explicitly to the startup function, that way you don't have to put a special override in the sumneko settings. Although I really wish that `sumneko` could somehow figure out that it is available within that scope.

Such as this?
```
return require('packer').startup(function(use,use_rocks)
```

Or do you mean something else?

**Comment by akinsho** [comment]
yep if you do
```lua
return require('packer').startup(function(use)
  use "plugin/thing.vim"
end)
```
That will silence `sumneko` without any lsp fiddling

**Comment by sandersantema** [comment]
> yep if you do
> 
> ```lua
> return require('packer').startup(function(use)
>   use "plugin/thing.vim"
> end)
> ```
> 
> That will silence `sumneko` without any lsp fiddling

Ah then I understood correctly, I was suggesting that it might be nice to add this to the default config in `README.MD` such as in the first snippet in my original issue message.

**Comment by wbthomason** [comment]
I'm OK adding a notice that you can silence the warnings by doing this, but - since the examples do work - I think I'd like to keep the default examples the same. Does that seem reasonable?

**Comment by sandersantema** [comment]
Seems like the best solution actually :+1:

**Comment by wbthomason** [comment]
Whoops, forgot that I resolved this already in 02c3627a
